intel-analytics/BigDL
# bugfix - set mask for container

## What changes were proposed in this pull request?
Add an argument 'recursiveFlag' in setMask series api
when set to true, it would set dimension mask for this module and all of its submodules,
otherwise set dimension mask only to itself.

P.S. 
In the previous version, the setMask api include recursion, and 'recursiveFlag' is false by default,
therefore I need to set the flag to true for existing code. 

## How was this patch tested?
For setInputDimMask, setOutputDimMask & setWeightDimMask
1. create a container module (a Sequential Module in my testcase)
2. add 2 non-container module (Convolution module)
3. Verify the mask value of Sequential and Convolution module inside is the default value 0
4. Modify Sequential's mask by calling set*XXX*DimMask functions with recursiveFlag off, verify only the Sequential's mask changed, and submodules remain unchanged
5. Make set*XXX*DimMask call on Sequential Module with recursive flag, verify all the modules including Sequential and Convolution modules are changed
6. Make set mask function calls to Convolution modules individually, verify only themselves are effected afterwards.

Jenkins result url:
http://mlt-bddl01.sh.intel.com:8080/job/BigDL-PRVN-lite/574/

## Related links or issues (optional)
fixed https://github.com/intel-analytics/BigDL/issues/2805
